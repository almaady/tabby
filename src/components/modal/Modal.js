// Modal
import React from "react";
import classNames from "classnames";
import PropTypes from "prop-types";

import ModalBackground from "./ModalBackground";
import ModalCard from "./ModalCard";

/**
 * @class Modal - It contains the modal card and the modal background
 */
class Modal extends React.PureComponent {
  /**
   * @property {bool} show - Determines if the modal is visible
   * @property {func} onClose - The function to close the modal
   * @property {node} children - The content of the modal
   * @property {string} className - Just in case you need another class
   */
  static propTypes = {
    show: PropTypes.bool.isRequired,
    onClose: PropTypes.func.isRequired,
    children: PropTypes.node.isRequired,
    className: PropTypes.string,
    classNameWrapper: PropTypes.string
  };

  render() {
    const { show, onClose, children, className, classNameWrapper } = this.props;
    return (
      <div className={classNames({ hidden: !show }, "modal--wrapper", classNameWrapper)}>
        <ModalBackground onClose={onClose} />
        <ModalCard onClose={onClose} className={className}>
          {children}
        </ModalCard>
      </div>
    );
  }
}
Modal.defaultProps = {
  className: "",
  classNameWrapper: ""
};
export default Modal;
