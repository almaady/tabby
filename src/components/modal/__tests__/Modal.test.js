import React from "react";
import { shallow, mount } from "enzyme";

import Modal from "../Modal";

describe("Modal", () => {
  let props;
  beforeEach(() => {
    props = {
      show: false,
      // eslint-disable-next-line react/jsx-filename-extension
      children: <div />,
      onClose: () => {}
    };
  });
  it("should render correctly with default state", () => {
    const component = shallow(<Modal {...props} />);
    expect(component).toMatchSnapshot();
  });

  it("should add hidden class when show is false", () => {
    const component = shallow(<Modal {...props} />);
    expect(component.hasClass("hidden")).toEqual(true);
  });

  it("should remove hiddden class when show is true", () => {
    const propsVisible = { ...props };
    propsVisible.show = true;
    const componentVisible = shallow(<Modal {...propsVisible} />);
    expect(componentVisible.hasClass("hidden")).toEqual(false);
  });
  it("should render correct onClose prop on ModalBackground", () => {
    const ModalBackground = shallow(<Modal {...props} />).find(
      "ModalBackground"
    );
    expect(ModalBackground.props().onClose).toBeFunction();
  });
  it("should render the same onClose func on ModalBackground", () => {
    const ModalBackground = shallow(<Modal {...props} />).find(
      "ModalBackground"
    );
    expect(ModalBackground.props().onClose).toEqual(props.onClose);
  });
  it("should render correct onClose prop on ModalCard", () => {
    const ModalCard = shallow(<Modal {...props} />).find("ModalCard");
    expect(ModalCard.props().onClose).toBeFunction();
  });
  it("should render the same onClose func on ModalCard", () => {
    const ModalBackground = shallow(<Modal {...props} />).find("ModalCard");
    expect(ModalBackground.props().onClose).toEqual(props.onClose);
  });
  it("should has a new classname", () => {
    const component =  shallow(<Modal {...props} classNameWrapper="classname--wrapper"/>);
    expect(component.hasClass("classname--wrapper")).toBeTrue();
  });
});
