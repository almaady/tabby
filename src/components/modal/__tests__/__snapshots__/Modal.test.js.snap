// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render correctly with default state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal
    className=""
    classNameWrapper=""
    onClose={[Function]}
    show={false}
  >
    <div />
  </Modal>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ModalBackground
          onClose={[Function]}
        />,
        <ModalCard
          className=""
          onClose={[Function]}
        >
          <div />
        </ModalCard>,
      ],
      "className": "hidden modal--wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onClose": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div />,
          "className": "",
          "onClose": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ModalBackground
            onClose={[Function]}
          />,
          <ModalCard
            className=""
            onClose={[Function]}
          >
            <div />
          </ModalCard>,
        ],
        "className": "hidden modal--wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onClose": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div />,
            "className": "",
            "onClose": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
